app/controllers/annotation_categories_controller.rb

```diff
@@ -71,6 +75,7 @@ def update
 
     if @annotation_category.update(annotation_category_params)
       flash_message(:success, t('.success'))
+      render 'show', assignment_id: @assignment.id, id: @annotation_category.id
```

Comment:
This is because the refreshing of the UI should happen in cases where the criterion is updated.